arithmetization/src/main/java/net/consensys/linea/zktracer/container/stacked/StackedList.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Consensys Software Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package net.consensys.linea.zktracer.container.stacked;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import lombok.Getter;
+import lombok.experimental.Accessors;
+import net.consensys.linea.zktracer.container.ModuleOperation;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Implements a system of nested lists behaving as a single one, where the current context
+ * modification can transparently be dropped.
+ *
+ * @param <E> the type of elements stored in the list
+ */
+@Accessors(fluent = true)
+public class StackedList<E extends ModuleOperation> {
+  private final List<E> operationSinceBeginningOfTheConflation = new ArrayList<>();
+  @Getter private final List<E> thisTransactionOperation = new ArrayList<>();
+  private final CountOnlyOperation lineCounter = new CountOnlyOperation();
```

Comment:
`operationsOfExternalCircuits` or `operationsLedgerForExternalCircuits` or ...

Reply:
I don't follow you this. Which external circuits are you talking about ? This is really the lineCount of the StackedSet / StackedList. 

Rhe purpose of this is to kill a loop on all previous operations when querrying lineCount

Reply:
My understanding (maybe wrong ?) was that CountOnlyOperations was for stuff like precompiles etc ... which we delegate to external circuits. Is that not the case ?

Reply:
Not only. We use it too in StackedSet and StackedList to avoid a loop when line counting. 
We use it too in WCP to count the "potential additional rows".